conftest.py

```diff
@@ -33,7 +33,7 @@ def seed(request):
 )
 settings.register_profile(
     "ci",
-    deadline=None,
+    deadline=900000, # 15 mins, should be enough to identify the problematic test
```

Comment:
with 15mins I think we can still run into issues, for example if a test takes between 7 and 10 mins but we run it 300 times

Reply:
I would merge the rust runs and then lower this to 2 mins